src/libtriton/engines/symbolic/symbolicEngine.cpp

```diff
@@ -451,47 +452,49 @@ namespace triton {
 
 
       /* Prints symbolic expression with used references and symbolic variables */
-      std::ostream& SymbolicEngine::printSlicedExpressions(std::ostream& stream, const triton::engines::symbolic::SharedSymbolicExpression& expr, bool assert_) {
-        auto symVars = this->getSymbolicVariables();
+      std::ostream& SymbolicEngine::printSlicedExpressions(std::ostream& stream, const SharedSymbolicExpression& expr, bool assert_) {
         auto ssa = this->sliceExpressions(expr);
-
-        std::vector<usize> ids;
-        ids.reserve(std::max(symVars.size(), ssa.size()));
-        for (const auto& symVar : symVars) {
-          ids.push_back(symVar.first);
+        std::vector<usize> symExprs;
+        std::set<usize> symVars;
+        for (const auto& se : ssa) {
+          symExprs.push_back(se.first);
+          auto children = ast::childrenExtraction(se.second->getAst(), false, false);
+          for (const auto& child : children) {
+            if (child->getType() == ast::VARIABLE_NODE) {
+              auto id = reinterpret_cast<ast::VariableNode *>(child.get())->getSymbolicVariable()->getId();
+              symVars.insert(id);
+            }
+          }
```

Comment:
Is the code above only for searching symbolic variables? Why not use `ast::search(expr->getAst(), ast::VARIABLE_NODE)`?

Reply:
Done